lib/classes/CLI.js

```diff
@@ -242,14 +247,66 @@ class CLI {
 
     this.consoleLog('');
 
-    // print all the installed plugins
-    this.consoleLog(chalk.yellow.underline('Plugins'));
+    const externalSortedPlugins = this.loadedPlugins
+      .filter(plugin => plugin.isExternal)
+      .sort((plugin1, plugin2) => plugin1.constructor.name.localeCompare(plugin2.constructor.name));
 
-    if (this.loadedPlugins.length) {
-      const sortedPlugins = this.loadedPlugins.sort((plugin1, plugin2) =>
-        plugin1.constructor.name.localeCompare(plugin2.constructor.name)
-      );
-      this.consoleLog(sortedPlugins.map(plugin => plugin.constructor.name).join(', '));
+    if (externalSortedPlugins.length) {
+      // print all the installed plugins
+      this.consoleLog(chalk.yellow.underline('Plugins'));
+
+      this.consoleLog(externalSortedPlugins.map(plugin => plugin.constructor.name).join(', '));
+
+      this.consoleLog('');
+      this.consoleLog(chalk.yellow.underline('Commands by plugin'));
+      this.consoleLog('');
+
+      let pluginCommands = {};
+
+      // add commands to pluginCommands based on command's plugin
+      const addToPluginCommands = cmd => {
+        const pcmd = _.clone(cmd);
+
+        // remove subcommand from clone
+        delete pcmd.commands;
+
+        // check if a plugin entry is already present in pluginCommands. Use the
+        // existing one or create a new plugin entry.
+        if (pluginCommands[pcmd.pluginName]) {
+          pluginCommands[pcmd.pluginName] = pluginCommands[pcmd.pluginName].concat(pcmd);
+        } else {
+          pluginCommands[pcmd.pluginName] = [pcmd];
+        }
+
+        // check for subcommands
+        if ('commands' in cmd) {
+          Object.values(cmd.commands).forEach(d => {
+            addToPluginCommands(d);
+          });
+        }
+      };
+
+      // fill up pluginCommands with commands in loadedCommands
+      Object.values(this.loadedCommands).forEach(details => {
+        if (details.isExternal) {
+          addToPluginCommands(details);
+        }
+      });
+
+      // sort plugins alphabetically
+      pluginCommands = _(Object.entries(pluginCommands))
+        .sortBy(0)
+        .fromPairs()
+        .value();
+
+      Object.entries(pluginCommands).forEach(([plugin, details]) => {
+        this.consoleLog(plugin);
+        details.forEach(cmd => {
+          // display command usage with single(1) indent
+          this.displayCommandUsage(cmd, cmd.key.split(':').join(' '), 1);
+        });
+        this.consoleLog('');
+      });
     } else {
       this.consoleLog('No plugins added yet');
```

Comment:
I assume this is to be displayed, when no external plugins are configured (?)

I think we can skip this message (In current implementation it was never shown, as there were always some internal plugins)

Reply:
but in this case under 'Plugin', we do not list internal plugins, So this can be seen right ?

Reply:
Yes, and we don't want to be seen. Message is weird "No plugins added yet" (why yet?).

I think best way is to not show "Plugins" section at all